modules/oas-page-builder/src/services/pageBuilder.ts

```diff
@@ -15,12 +18,42 @@ const fetchTextData = async (url: string, errMsg: string) => {
   return res.text();
 };
 
+const createFetchGitHash = () => {
+  let gitHashCache: string;
+  return {
+    fetchGitHash: async () => {
+      if (gitHashCache) return gitHashCache;
+      try {
+        const gitHash = await fetchTextData(GIT_HASH_URL, 'Could not find current version or git hash');
+        gitHashCache = gitHash;
+        return gitHash;
+      } catch (e) {
+        throw new Error('Unsuccessful git hash fetch');
+      }
+    },
+    resetGitHashCache: () => {
+      gitHashCache = '';
+    },
+  };
+};
+
+const { fetchGitHash, resetGitHashCache } = createFetchGitHash();
+
 interface AtlasSpecUrlParams {
   apiKeyword: string;
   apiVersion?: string;
   resourceVersion?: string;
 }
 
+const ensureSavedVersionDataMatches = async (versions: VersionData, apiVersion?: string, resourceVersion?: string) => {
```

Comment:
NIT: Since we don't await anything, this function doesn't need to be async

Reply:
Thanks, left that in. My bad!